tests/scripts/scancode_plugins/src/notice_generator/notice_generator.py

```diff
@@ -0,0 +1,287 @@
+            else:
+                for original_requirement in req_data:
+                    cleaned_requirement_name: str = original_requirement.split("=")[0].strip(">").strip("\n")
+                    package_name: str = cleaned_requirement_name.replace("-", "_")
+
+                    if package_name not in self.required_packages:
+                        if "[" and "]" in package_name:
+                            package_name = package_name.split("[")[0]
+
+                        self.required_packages[package_name] = cleaned_requirement_name
+
+    def verify_license_in_packages(self, processed_package: str) -> None:
+        """
+        Checks if the license_content exists for all packages
+        If license not found, it builds the url for the raw github text and fetches the license content
+        """
+        if (
+            processed_package in self.processed_packages
+            and "license_content" not in self.processed_packages[processed_package]
+        ):
+            try:
+                raw_github_base_url: str = "https://raw.githubusercontent.com"
+                scancode_url: str = self.processed_packages[processed_package]["scancode_url"]
+                scancode_license_content_url: str = ""
+
+                if scancode_url:
+                    splitted_scancode_url: list[str] = scancode_url.split("/")
+
+                    if "tree" in splitted_scancode_url:
+                        splitted_scancode_url.remove("tree")
+
+                    scancode_license_content_url = f"{raw_github_base_url}/{'/'.join(splitted_scancode_url[3:])}"
+
+                response: Response = requests.get(scancode_license_content_url)
+
+                if response.status_code == 200:
+                    self.processed_packages[processed_package]["license_content"] = response.text
+                    self.processed_packages[processed_package]["license_path"] = scancode_url
+                else:
+                    print(f"License could not be found at: {scancode_url}")
+
+            except Exception as e:
+                raise e
+
+    def write_to_notice_file(self, package_data: dict[str, str]) -> None:
+        """
+        Writes the NOTICE.txt file with the package data
+        """
+        package_name = package_data["package_name"]
+        package_version = package_data.get("version")
+        package_homepage_url = package_data.get("homepage_url")
+        package_license_name = package_data["license_name"]
+        package_license_path = package_data.get("license_path")
+        package_license_content = package_data.get("license_content")
+
+        if not package_version:
+            package_version = ""
+
+        if not package_homepage_url:
+            package_homepage_url = ""
+
+        if not package_license_path:
+            package_license_path = ""
+
+        if not package_license_content:
+            package_license_content = ""
+
+        with open(self.notice_file_name, "a+") as fh:
```

Comment:
this should be better to move to a default scancode template/output on the long run